src/test/java/org/vandeseer/easytable/util/PdfUtilTest.java

```diff
@@ -86,4 +99,20 @@ public void getOptimalTextBreakLines_noSpacesNorDotsNorCommasInText_shouldSplitB
         assertThat(lines.get(2), is(equalTo("lySoWeBreakBySize")));
     }
 
+
+    @Test
+    public void testVeryBigText() {
+        StringBuilder builder = new StringBuilder();
+
+        for (int i = 0; i < 50; i++) {
+            builder.append("https://averylonginternetdnsnamewhich-maybe-breaks-easytable.com ");
+        }
+
+        System.out.println("Got these lines back:");
+        PdfUtil.getOptimalTextBreakLines(builder.toString(), PDType1Font.HELVETICA, 8, 102)
+                .forEach(System.out::println);
```

Comment:
Please use one or several asserts here, otherwise we won't be able to detect regression automatically. 

E.g. like so: 

        assertThat(optimalTextBreakLines.size(), equalTo(150));

        for (int i = 0; i < optimalTextBreakLines.size(); i++) {
            if (i % 3 == 0) {
                assertThat(optimalTextBreakLines.get(i), equalTo("https://averylonginternetdns-"));
            } else if (i % 3 == 1) {
                assertThat(optimalTextBreakLines.get(i), equalTo("namewhich-maybe-breaks-"));
            } else if (i % 3 == 2) {
                assertThat(optimalTextBreakLines.get(i), equalTo("easytable.com"));
            }
        }

This is still not optimal, because it uses control flow in a test case and also several asserts per test case – but it's ok for this purpose I'd say ;)

Reply:
You're right. The other problem (which occurs without my fix) is that the method runs forever. 
So I will add your hints and a test-timeout (normally I dislike them, because runTime can change on different environments).